packages/plugin-ext/src/plugin/terminal-ext.ts

```diff
@@ -138,9 +138,10 @@ export class TerminalServiceExtImpl implements TerminalServiceExt {
         }
     }
 
-    $terminalClosed(id: string): void {
+    $terminalClosed(id: string, exitStatus: theia.TerminalExitStatus | undefined): void {
         const terminal = this._terminals.get(id);
         if (terminal) {
+            terminal.updateExitStatus(exitStatus);
```

Comment:
Just directly write to the public field?

I also feel we should handle the case where `exitStatus` is `undefined`: in that case we should set `exitStatus` to `{ code: undefined }`.

Reply:
VScode's API states that this field is `readonly`, this is why I defined get and set methods in the implementation. This might be the wrong way of dealing with it though, this is something I hadn't done in Typescript before.

Regarding the handling of `exitStatus` being `undefined`, that could be a smarter workaround than what I had proposed for `TerminalWidgetImpl#OnTermDidClose`.

Reply:
Resolved in commit : [Fixed exitStatus undefined when closing terminal manually](https://github.com/eclipse-theia/theia/pull/11175/commits/962e252d55d4608f218968bd2be5e7dfcae304f4)